src/embed_tests/Codecs.cs

```diff
@@ -83,4 +83,33 @@ public class Codecs {
             }
         }
     }
+
+    class FakeEncoder<T> : IPyObjectEncoder
+    {
+        public bool CanEncode(Type type) => type == typeof(T);
+        public PyObject TryEncode(object value) => this.GetRawPythonProxy();
+    }
+
+    class FakeDecoder<TTarget> : IPyObjectDecoder
```

Comment:
Maybe give this a few lines of documentation to make it easier to follow the tests.

Reply:
@filmor renamed classes to make their behavior more clear, and added xmldoc.